core/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/repair/OnDemandRepairSchedulerImpl.java

```diff
@@ -68,6 +71,10 @@ private OnDemandRepairSchedulerImpl(Builder builder)
         myRepairConfiguration = builder.repairConfiguration;
         myRepairHistory = builder.repairHistory;
         myExecutor.scheduleWithFixedDelay(() -> clearFailedJobs(), DEFAULT_INITIAL_DELAY_IN_DAYS, DEFAULT_DELAY_IN_DAYS, TimeUnit.DAYS);
+        myOndemandStatus = new OnDemandStatus(builder.nativeConnectionProvider);
```

Comment:
Wouldn't it be better to create this outside of the scheduler and pass it in instead of nativeConnectionProvider?